### PR TITLE
Update contributors guide

### PR DESCRIPTION
**Pull request by e-korolevskii**
Updated contributors guide
- [ ] Mark if documentation changes are required.

### CODE REVIEW

File: docs/contributors.md [hunk 1]
```diff
@@ -1,40 +1,113 @@
 # Contributors
 
-Thank you for contributing! This action is targetted around setting up the dotnet cli and related sdks for GitHub actions. As part of that we use proxy settings (for self-hosted runners) and set-up nuget authentication for private feeds.
+Thank you for contributing!
```
Comment (dmitry-shibanov) [inline docs/contributors.md:3]: I think we can return description of the action.

File: docs/contributors.md [hunk 2]
```diff
@@ -1,40 +1,113 @@
...
 
-For example, if you're publishing v2:
-```
-git tag -fa v2 -m "Update v2 tag"
-git push origin v2 --force
-```
\ No newline at end of file
+**How to contribute:**
+
+Pull requests are the easiest way to contribute changes to git repos at GitHub. They are the preferred contribution method, as they offer a nice way of commenting and amending the proposed changes.
```
Comment (dmitry-shibanov) [inline docs/contributors.md]: `nice way` to `convenient way`.

File: docs/contributors.md [hunk 3]
```diff
@@ -1,40 +1,113 @@
...
+
+- Please check that no one else has already created a pull request with these changes
```
Comment (dmitry-shibanov) [inline docs/contributors.md]: `with these changes` to `with these or similar changes`

File: docs/contributors.md [hunk 4]
```diff
@@ -1,40 +1,113 @@
...
+- Use a "feature branch" for your changes. That separates the changes in the pull request from your other changes and makes it easy to edit/amend commits in the pull request 
+- Make sure your changes are formatted correctly and consistently with the rest of the documentation
+- Re-read what you wrote, and run a spellchecker on it to make sure you didn't miss anything
```
Comment (dmitry-shibanov) [inline docs/contributors.md:30]: Do we need to mention it ?

File: docs/contributors.md [hunk 5]
```diff
@@ -1,40 +1,113 @@
...
+- If your pull request is connected to an open issue, please, leave a link to this issue in the `Related issue:` section
+- If you later need to add new commits to the pull request, you can simply commit the changes to the local branch and then push them. The pull request gets automatically updated
+
+**Once you've filed the pull request:**
+
+- Maintainers will review your pull request
+- If a maintainer requests changes, first of all, try to think about this request critically and only after that implement and request another review
+- If your PR gets accepted, it will soon be merged into the main branch. But your contribution will take effect only after the release of a new version of the action
```
Comment (dmitry-shibanov) [inline docs/contributors.md]: `But your contribution will take effect only after the release of a new version of the action` to `But your contribution will take effect only after the release of a new version of the action and updating the major tag`

File: docs/contributors.md [hunk 6]
```diff
@@ -1,40 +1,113 @@
...
+
+We like code commits a lot! They're super handy, and they keep the project going and doing the work it needs to do to be useful to others.
+
+Code contributions of just about any size are acceptable!
+
+The main difference between code contributions and documentation contributions is that contributing code requires the inclusion of relevant tests for the code being added or changed. Contributions without accompanying tests will be held off until a test is added unless the maintainers consider the specific tests to be either impossible or way too much of a burden for such a contribution.
+
+**How to contribute:**
+
+Pull requests are the easiest way to contribute changes to git repos at GitHub. They are the preferred contribution method, as they offer a nice way of commenting and amending the proposed changes.
```
Comment (dmitry-shibanov) [inline docs/contributors.md]: the same as for previous.

File: docs/contributors.md [hunk 7]
```diff
@@ -1,40 +1,113 @@
...
+**Learn more about how to work with the repository:**
+
+- To implement new features or fix bugs, you need to make changes to the `.ts` files, which are located in the `src` folder
+- To comply with the code style, **you need to run the `format` script**
+- To transpile source code to `javascript` we use [NCC](https://github.com/vercel/ncc). **It is very important to run the `build` script after making changes**, otherwise your changes will not get into the final `javascript` build
+
+**Learn more about how to implement tests:**
+
+Adding or changing tests is an integral part of making a change to the code. 
+Unit tests are in the `__tests__` folder, and end-to-end tests are in the `workflows` folder (in particular, in the file [test-dotnet.yml](https://github.com/actions/setup-dotnet/blob/main/.github/workflows/test-dotnet.yml).
```
Comment (dmitry-shibanov) [inline docs/contributors.md]: `in particular, in the file` to `in the particular file`

File: docs/contributors.md [hunk 8]
```diff
@@ -1,40 +1,113 @@
...
-git push origin v2 --force
-```
\ No newline at end of file
+**How to contribute:**
+
+Pull requests are the easiest way to contribute changes to git repos at GitHub. They are the preferred contribution method, as they offer a nice way of commenting and amending the proposed changes.
+
+- Please check that no one else has already created a pull request with these changes
+- Use a "feature branch" for your changes. That separates the changes in the pull request from your other changes and makes it easy to edit/amend commits in the pull request 
+- Make sure your changes are formatted correctly and consistently with the rest of the documentation
```
Comment (marko-zivic-93) [inline docs/contributors.md]: formatted correctly -> formatted properly

File: docs/contributors.md (hunk 4 above)
Comment (marko-zivic-93) [reply docs/contributors.md:30]: Why not leave it as is. We should encourage people to check their spelling :)

File: docs/contributors.md (hunk 7 above)
Comment (marko-zivic-93) [reply docs/contributors.md]: I would go with "to be exact, in the file ..."

File: docs/contributors.md [hunk 9]
```diff
@@ -1,40 +1,113 @@
...
+
+- Avoid responding to issues you don't know you can answer accurately
+- Try to refer to past issues with accepted answers as much as possible. Link to them from your replies
+- Be kind and patient with users. Often, folks who have run into confusing things might be upset or impatient. This is natural. If you feel uncomfortable in conversation with them, it's better to stay away or withdraw from the issue.
+
+ > If some user is violating our code of conduct [standards](https://github.com/actions/setup-dotnet/blob/main/CODE_OF_CONDUCT.md#our-standards), refer to the [Enforcement](https://github.com/actions/setup-dotnet/blob/main/CODE_OF_CONDUCT.md#enforcement) section of the Code of Conduct to resolve the conflict
+ 
+## Review pull requests
+ 
+Another great way to contribute is pull request reviews. Please, be extra kind: people who submit code/doc contributions are putting themselves in a pretty vulnerable position, and have put time and care into what they've done (even if that's not obvious to you!) Please, always respond with respect, and be understanding, but don't feel like you need to sacrifice your standards for their sake, either.
```
Comment (marko-zivic-93) [inline docs/contributors.md]: Another great way to contribute is pull request reviews. => Another great way to contribute is to review pull request.

**Review by marko-zivic-93 — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Review by vsafonkin — APPROVED** [review]

File: docs/contributors.md [hunk 10]
```diff
@@ -1,40 +1,113 @@
...
+**Learn more about how to work with the repository:**
+
+- To implement new features or fix bugs, you need to make changes to the `.ts` files, which are located in the `src` folder
+- To comply with the code style, **you need to run the `format` script**
+- To transpile source code to `javascript` we use [NCC](https://github.com/vercel/ncc). **It is very important to run the `build` script after making changes**, otherwise your changes will not get into the final `javascript` build
+
+**Learn more about how to implement tests:**
+
+Adding or changing tests is an integral part of making a change to the code. 
+Unit tests are in the `__tests__` folder, and end-to-end tests are in the `workflows` folder (in the particular file [test-dotnet.yml](https://github.com/actions/setup-dotnet/blob/main/.github/workflows/test-dotnet.yml).
```
Comment (IvanZosimov) [inline docs/contributors.md]:
```suggestion
Unit tests are in the `__tests__` folder, and end-to-end tests are in the `workflows` folder, particularly in the [workflow.yml](https://github.com/actions/setup-dotnet/blob/main/.github/workflows/workflow.yml).
```

Comment (IvanZosimov) [reply docs/contributors.md]: Wrong file

**Review by IvanZosimov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request closed without merging**